packages/pcd/gpc-pcd/src/GPCPCDPackage.ts

```diff
@@ -0,0 +1,305 @@
+  if (!serializedIdentityPCD) {
+    throw new Error("Cannot make proof: missing owner PCD");
+  }
+  const ownerSemaphorePCD =
+    serializedIdentityPCD !== undefined
+      ? await SemaphoreIdentityPCDPackage.deserialize(serializedIdentityPCD)
+      : undefined;
+
+  const externalNullifier =
+    args.externalNullifier.value !== undefined
+      ? ({
+          type: "string",
+          value: args.externalNullifier.value
+        } satisfies PODStringValue)
+      : undefined;
+  if (externalNullifier !== undefined && ownerSemaphorePCD === undefined) {
+    throw new Error("External nullifier requires an owner identity PCD.");
+  }
+  const watermark =
+    args.watermark.value !== undefined
+      ? ({
+          type: "string",
+          value: args.watermark.value
+        } satisfies PODStringValue)
+      : undefined;
+
+  return {
+    proofConfig,
+    proofInputs: {
+      pods: { pod0: podPCD.pod },
+      ...(ownerSemaphorePCD !== undefined
+        ? {
+            owner: {
+              semaphoreV3: ownerSemaphorePCD?.claim?.identity,
+              externalNullifier: externalNullifier
+            }
+          }
+        : {}),
+      watermark: watermark
+    }
+  };
+}
+
+/**
+ * Creates a new {@link GPCPCD} by generating an {@link GPCPCDProof}
+ * and deriving an {@link GPCPCDClaim} from the given {@link GPCPCDArgs}.
+ *
+ * This generates a ZK proof using the given config and inputs using a
+ * selected circuit from the supported family.
+ *
+ * @throws if the arguments are invalid
+ */
+export async function prove(args: GPCPCDArgs): Promise<GPCPCD> {
+  const zkArtifactPath = ensureInitialized();
+  const { proofConfig, proofInputs } = await checkProofArgs(args);
+  const id =
+    args.id !== undefined && typeof args.id.value === "string"
+      ? args.id.value
+      : uuid();
+
+  const { boundConfig, revealedClaims, proof } = await gpcProve(
+    proofConfig,
+    proofInputs,
+    zkArtifactPath
+  );
+
+  return new GPCPCD(
+    id,
+    {
+      config: boundConfig,
+      revealed: revealedClaims
+    },
+    { groth16Proof: proof }
+  );
+}
+
+/**
+ * Verifies a POD PCD by checking that its {@link GPCPCDClaim} corresponds to
+ * its {@link GPCPCDProof}.
+ *
+ * This confirms that the claimed configuration and revealed inputs match
+ * what was used to prove.
+ */
+export async function verify(pcd: GPCPCD): Promise<boolean> {
+  const zkArtifactPath = ensureInitialized();
+  try {
+    return gpcVerify(
+      pcd.proof.groth16Proof,
+      pcd.claim.config,
+      pcd.claim.revealed,
+      zkArtifactPath
+    );
+  } catch (e) {
+    console.error("Verifying invalid GPC proof data:", e);
+    return false;
+  }
+}
+
+/**
+ * Serializes a {@link GPCPCD}.
+ *
+ * @param pcd The GPC PCD to be serialized.
+ * @returns The serialized version of the GPC PCD.
+ */
+export async function serialize(pcd: GPCPCD): Promise<SerializedPCD<GPCPCD>> {
+  return {
+    type: GPCPCDTypeName,
+    pcd: JSON.stringify({
```

Comment:
This had me worried at first, as JSON.stringify can't cope with bigint values. However, since the `config` and `revealed` components of the `claim` are separately serialized as strings, this is safe. Might be worth a comment to that effect, to ensure that nobody breaks this assumption in future.

Reply:
Yeah, I need a more consistent strategy for serialization.  I've thought about making a separate set of JSON-compatible object types (like JSONProofConfig).  Or just giving up and stringifying the bigints all the time.  It's something which needs a more detailed look than I have time for in this PR.

Will clarify with a comment for now.